fix(trace): Handle multiple perf issues in a trace

wmak opened the pull request:
- This fixes a bug where we weren't handling multiple perf issues in a single trace since we did it by mapping event_ids to occurrence_ids, but there could be multiple occurrence_id for a given event_id
- This also fixes a bug (only with new spans view) where the list of suspect spans was _all_ the suspect spans across all perf issues for every perf issue, rather than just the suspect spans for a given perf issue
- Snuck some decorators to add more tracing too

wmak committed:
fix(trace): Handle multiple perf issues in a trace

- This fixes a bug where we weren't handling multiple perf issues in a
  single trace since we did it by mapping event_ids to occurrence_ids,
  but there could be multiple occurrence_id for a given event_id
- This also fixes a bug (only with new spans view) where the list of
  suspect spans was _all_ the suspect spans across all perf issues for
  every perf issue, rather than just the suspect spans for a given perf
  issue

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

k-fish commented on src/sentry/api/endpoints/organization_events_trace.py:
```diff
@@ -189,6 +192,64 @@ def nodestore_event(self) -> Event | None:
+                        else:
+                            end = max(end, end_timestamp)
+                        suspect_spans.append(event_span.get("span_id"))
+            # Logic for qualified_short_id is copied from property on the Group model
+            # to prevent an N+1 query from accessing project.slug everytime
+            qualified_short_id = None
+            project_slug = self.event["project"]
+            if group.short_id is not None:
+                qualified_short_id = f"{project_slug.upper()}-{base32_encode(group.short_id)}"
```
So we know that this project slug is unique per this `TraceEvent`, maybe instead allow `qualified_short_id` on the Group model to be passed a project_id to skip n+1, and mention this on the model instead so people can benefit in the future?

wmak replied:
I'll do this in a follow up

k-fish reviewed (approved):
You have a few tests that a breaking, but seems related to `suspect_spans` as mentioned, once those are fixed, lgtm

wmak committed:
fix: Actually only associate the same group back to its perf issue

- Because we didn't have a map from occurrence ids to issues we just
  slammed all the occurrences together under a given group id

wmak committed:
fix: I didn't like that we were iterating on the groups

- iterate on the occurrences instead since each one of those should
  represent 1 tangible performance issue happening

wmak commented on src/sentry/api/endpoints/organization_events_trace.py:
```diff
@@ -189,6 +192,64 @@ def nodestore_event(self) -> Event | None:
+                    "start": start,
+                    "end": end,
+                }
+            )
```
as discussed in slack, looks like this is completely wrong lol. Re did my entire approach :+1:
